Encode provenance information into PURLs

sschuberth opened the pull request:
Please have a look at the individual commit messages for the details.

Etsija reviewed (commented):
The changes look OK, but how should we deploy this?  Meaning, when this is merged, every subsequent scan job (Jenkins etc.) will update the "bookmarking" for every scanned project and dependencies.  That is all and well, but until DOS package configuration provider is also changed, the evaluator will not be able to use those PURLs with extra data.

This leaves us with a (hopefully short) time period of "failing" Jenkins results for Evaluator.

sschuberth commented:
> but until DOS package configuration provider is also changed

Good point, I've added a commit to also use the new PURLs with full provenance information in the package curation provider.

> the evaluator will not be able to use those PURLs with extra data.

I'll now look into getting the PURL links right also in the how-to-fix hints created in the evaluator's rules.

sschuberth commented:
> I'll now look into getting the PURL links right also in the how-to-fix hints created in the evaluator's rules.

See https://github.com/doubleopen-project/ort-config/pull/17.
